src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -146,6 +152,36 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized int getJvmId() {
+        try {
+            if (!isConnected()) {
+                connect();
+            }
+            ObjectName runtimeBean = new ObjectName("java.lang:type=Runtime");
+            List<String> attributes =
+                    new ArrayList<>(
+                            Arrays.asList(
+                                    "ClassPath",
+                                    "Name",
+                                    "InputArguments",
+                                    "LibraryPath",
+                                    "VmVendor",
+                                    "VmVersion",
+                                    "StartTime"));
+            HashCodeBuilder jvmId = new HashCodeBuilder();
+
+            for (String attr : attributes) {
+                jvmId.append(
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, runtimeBean, attr)));
+            }
+            return jvmId.hashCode();
+        } catch (Exception e) {
+            cw.println(e);
+            return 0;
```

Comment:
I think I'd prefer to re-throw than to return 0 here. This would swallow connection exceptions occurring because the target JVM has gone offline, for example, and we'd end up mapping all such JVMs to the same 0 ID.